x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -403,6 +403,9 @@ private MethodSpec addRawVector(boolean masked) {
             if (masked) {
                 builder.beginControlFlow("if (mask.getBoolean(i) == false)").addStatement("continue").endControlFlow();
             }
+            if (stateTypeHasSeen && masked) {
+                builder.addStatement("state.seen(true)");
+            }
```

Comment:
Another option is to build an `anyTrue` and stick it on `BooleanVector`. Then we'd check it early and bail if none are true. This'd be different than the `isConstant()` dance we do now in that it is allowed to check even. That check feels like it'd be faster than the shifted `state.seen` thing that I'm doing here....

I think I've convinced myself that's a better fix. What do you think, @ivancea?

Reply:
I think I'll expand the microbenchmarks in the morning and check. Seems simple enough.